src/Our.Umbraco.EnvironmentDashboard/Components/EnvironmentDashboardGroupsCollection.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System.Collections.Generic;
+using Umbraco.Core.Composing;
+
+namespace Our.Umbraco.EnvironmentDashboard.Components
+{
+	public class EnvironmentDashboardGroupsCollection : BuilderCollectionBase<IDashboardGroupsProvider>
+	{
+		public EnvironmentDashboardGroupsCollection(IEnumerable<IDashboardGroupsProvider> items) : base(items)
+		{
```

Comment:
I'm not entire sure what this is for, as I haven't used it. So I'm not sure why it needs to be Group**s**, and not just singular Group

The words Environment and Dashboard appear a lot, which can cause some confusion, so I would probably rename EnvironmentDashboardGroupsCollectionBuilder to something like InfoGroupCollection

Reply:
I named it Groups it should actually be EnvironmentDashboardGroupsProviderCollection

I made the GroupsProvider return a list of groups as I don't like returning null's and filtering null values.

null should't be used for logic in my eyes

Reply:
Perhaps name this InfoGroupProviderCollection ?

I'm still unsure why a collection of these providers is required :(

Reply:
> > I have made a DefaultEnvironmentDetector that looks for and AppSetting called "DashboardEnvironment"
> > The DomainEnvironmentDetector configuration looks like this now:
> > ```
> > 		composition.AddEnvironmentDashboard(environments =>
> > 			{
> > 				environments
> > 					.AddEnvironment("Local", "localhost:21801")
> > 					.AddEnvironment("Development", "environmentdashboard-dev.ry.com")
> > 					.AddEnvironment("QA", "environmentdashboard-qa.ry.com")
> > 					.AddEnvironment("UAT", "environmentdashboard-uat.ry.com");
> > 			})
> > 			.Append<UptimeDashboardGroupsProvider>()
> > 			.Append<DatabaseServerGroupsProvider>();
> > ```
> 
> Can we get away with no app setting? I prefer to avoid them as they become "yet another" app setting to potentially have to transform.
> 
> AddEnvironment(...) looks good! I guess we could even have an optional overload which takes an IEnumerable for the cases were people have multiple domains per environment.

that overload is already there the first two arguments are required, but there is an optional params string[] domains in the method aswell :)